addons/account/wizard/account_invoice_refund.py

```diff
@@ -113,9 +114,9 @@ def compute_refund(self, cr, uid, ids, mode='refund', context=None):
             journal_id = form.journal_id.id
             for inv in inv_obj.browse(cr, uid, context.get('active_ids'), context=context):
                 if inv.state in ['draft', 'proforma2', 'cancel']:
-                    raise osv.except_osv(_('Error!'), _('Cannot %s draft/proforma/cancel invoice.') % (mode))
```

Comment:
Will give weird messages in non-english translation

`Ne peut pas refund les factures brouillon/proforma/annulées`

Use a generic phrase, i.e
`Cannot operate on draft/proforma/cancel invoices.`

Same 2 lines below